context-free.md

```diff
@@ -2,21 +2,61 @@
 
 ##  Who is this programming language for?
 
+Context Free is for artists iterested in exploring generative art.
 
 ## What is easy to do in this language? Why is it easy?
 
+It's easy to generate art that has a lot of symmetry. One of the easiest tricks
+for generating an image is to create recursive shape rules, and this tends to
+lead to shapes with a recursive structure, which results in symmetry. The
+handful of tree-like example programs confirm this.
 
 ## What is hard to do in this language? Why is it hard?
 
+It would be hard to replicate a picture of some real object in the language.
+Encoding the information in a picture (which probably does not have too much
+symmetry) would take a lot of code in this language, whose primary strength is
+encoding symmetry in only a few lines.
 
-## How did you learn how to program in this language?
+Obviously, it would also be tremendously difficult to do any sort of
+computation or general-purpose programming in this language (although not
+impossible; the language has conditionals, loops, variables, functions, and
+recursion, so I believe it's Turing-complete).
 
+## How did you learn how to program in this language?
+I first read the documentation and tried to form a mental model for how the
+language works. Then I looked at the example programs, and tried to understand
+how they work, tweak things, and then change my mental model based on the
+results. Finally, I started writing a program and experimenting.
 
 ## What is the underlying _computational model_ for this programming language? 
 _We don't yet have a great definition of the term "computational model". 
 For now, try to come up with the clearest, most concise explanation of what 
 happens when a ContextFree program runs._
 
+Computation begins with the `startshape`. If this is a primitive, the shape is
```

Comment:
I learned a lot from reading Jeb's speculations about the underlying computational model of ContextFree. My guess was much simpler.